ethers-core/src/types/transaction/eip2718.rs

```diff
@@ -417,7 +417,7 @@ impl TypedTransaction {
 impl Decodable for TypedTransaction {
     fn decode(rlp: &rlp::Rlp) -> Result<Self, rlp::DecoderError> {
         let tx_type: Option<U64> = match rlp.is_data() {
-            true => Some(rlp.data()?.into()),
+            true => Some(U64::decode(rlp)?),
```

Comment:
hmm, unclear what this even did before.

this entire function looks a bit weird.

this should not rlp decode here I believe, because this is the type id of the envelope?

unclear what `rlp.data()?.into()` even did, but I think this here should just use the single byte.

idk what `is_data` is

Reply:
`rlp.data()` returns `Result<&[u8]>`, and apparently Parity Uint types [implement `From<&[u8]>`](https://docs.rs/uint/latest/src/uint/uint.rs.html#1363-1368) [by panicking](https://docs.rs/uint/latest/src/uint/uint.rs.html#1239-1253) ^^

Reply:
I refactored a bit to explicitly check the first byte as a the tx type - see [e7f8369](https://github.com/gakonst/ethers-rs/pull/2694/commits/e7f836907980e9de4cf4a4d83cecdd34dd381258).

It is a bit strange that the initial code was trying to coerce into a `U64`, when EIP-2718 specifies a single u8 as the tx type

Reply:
I also noticed [`decode_signed` has similar logic ](https://github.com/gakonst/ethers-rs/blob/f0e5b194f09c533feb10d1a686ddb9e5946ec107/ethers-core/src/types/transaction/eip2718.rs#L381-L413) and saw that you made a similar refactor to it in [pr 1733](https://github.com/gakonst/ethers-rs/pull/1733/files)